catalog/nsili/catalog-nsili-endpoint/src/main/java/com/connexta/alliance/nsili/endpoint/requests/GetRelatedFilesRequestImpl.java

```diff
@@ -64,17 +158,54 @@ public void cancel() throws ProcessingFault, SystemFault {
     @Override
     public String register_callback(Callback acallback)
             throws InvalidInputParameter, ProcessingFault, SystemFault {
-        return "";
+        throw new NO_IMPLEMENT("GetRelatedFilesRequest doesn't support callbacks");
     }
 
     @Override
     public void free_callback(String id)
             throws InvalidInputParameter, ProcessingFault, SystemFault {
-        return;
     }
 
     @Override
     public RequestManager get_request_manager() throws ProcessingFault, SystemFault {
         return new _RequestManagerStub();
     }
+
+    private String storeThumbnail(Metacard metacard) throws IOException, MimeTypeException {
+        String id = metacard.getId();
+
+        byte[] thumbnailBytes = metacard.getThumbnail();
+        TikaInputStream tis = TikaInputStream.get(thumbnailBytes);
+        MediaType mediaType = DETECTOR.detect(tis, new Metadata());
+        MimeType mimeType = TikaConfig.getDefaultConfig()
+                .getMimeRepository()
+                .forName(mediaType.toString());
+
+        String fileName = id + "-THUMBNAIL" + mimeType.getExtension();
+
+        String urlStr =
+                DEFAULT_PROTOCOL + "://" + location.host_name + (port == null ? "" : ":" + port)
+                        + location.path_name + "/" + fileName;
+
+        LOGGER.debug("Storing thumbnail for {} at location: {}", metacard.getTitle(), urlStr);
+
+        HttpPut httpPut = new HttpPut(urlStr);
+        HttpEntity entity = new ByteArrayEntity(thumbnailBytes);
+        httpPut.setEntity(entity);
+
+        Header contentTypeHeader = new BasicHeader("Content-Type", mediaType.toString());
+        httpPut.addHeader(contentTypeHeader);
+        HttpResponse response = httpClient.execute(httpPut);
+        int statusCode = response.getStatusLine().getStatusCode();
+        if (! (statusCode == 200 || statusCode == 201 || statusCode == 202 || statusCode == 204)) {
```

Comment:
Nit, but I think you can use HttpStatus.SC_OK, HttpStatus.SC_CREATED, HttpStatus.SC_ACCEPTED, and HttpStatus.SC_NO_CONTENT here